lib/AST/USRGeneration.cpp

```diff
@@ -226,7 +226,7 @@ bool ide::printDeclUSR(const ValueDecl *D, raw_ostream &OS) {
     return printObjCUSR(D, OS);
   }
 
-  if (!D->hasInterfaceType())
+  if (!D->hasInterfaceType() || D->getInterfaceType()->is<ErrorType>())
```

Comment:
This is the reason the test is filing, could you verify whether this change is necessary, is it really fixing a crash in your original test case ?